src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs

```diff
@@ -126,15 +126,25 @@ void ConfigureListenOptions(ListenOptions listenOptions)
     [Fact]
     public async Task HandshakeDetailsAreAvailable()
     {
+        string expectedHostname = null;
         void ConfigureListenOptions(ListenOptions listenOptions)
         {
-            listenOptions.UseHttps(new HttpsConnectionAdapterOptions { ServerCertificate = _x509Certificate2 });
+            listenOptions.UseHttps(
+                new HttpsConnectionAdapterOptions
+                {
+                    ServerCertificateSelector = (connection, name) =>
+                    {
+                        expectedHostname = name;
```

Comment:
Isn't it always 'localhost'? Or does this cover older platforms that don't supply it?

Reply:
Its always 'localhost' when I run the tests locally, but yes, I wanted to generalize it to cover all platforms and possible future code changes.